src/lib/tls/asio/asio_async_ops.h

```diff
@@ -149,6 +149,9 @@ class AsyncReadOperation : public AsyncBase<Handler, typename Stream::executor_t
                   m_stream.native_handle()->received_data(
                      static_cast<const uint8_t*>(read_buffer.data()), read_buffer.size()
                   );
+
+                  if (m_stream.native_handle()->is_closed())
+                     { ec = boost::asio::error::eof; }
```

Comment:
`native_handle()->is_closed()` tells means `received_data` has processed a `close_notify` alert and closed the channel. We set the EOF error to communicate this. Below, `map_error_code` might change that EOF to `StreamTruncated` (short read).
Can we make the assumption that we received `close_notify` from `is_closed()` here? Do I need to add more comments here?

In addition, `received_data` will have put the corresponding `close_notify` response into the send buffer.